cli/src/commands/diffedit.rs

```diff
@@ -59,6 +59,9 @@ pub(crate) struct DiffeditArgs {
     /// Specify diff editor to be used
     #[arg(long, value_name = "NAME")]
     tool: Option<String>,
+    /// Do not modify the content of the revision specified with `-r` or `--to`
+    #[arg(long)]
+    preserve_descendants: bool,
```

Comment:
nit: `--preserve-descendants` sounds like the default behavior is to somehow destroy descendants. `--reparent-descendants` or `--preserve-descendant-snapshots` ("snapshots" compared to "diffs") might be slightly better.

Reply:
I feel like `--preserve-descendant-snapshots` is too long to be usable and too technical. Even `content` would be clearer than `snapshots` for the end-user I think.

Reply:
I agree that `--preserve-descendant-snapshots` is too technical (not sure I think it's too long because users will probably set up shell completion). I also agree with Yuya's concern. I'm not sure what's a better name. Another way of viewing it is that we apply the reverse change in the descendants, but I suspect it would be even more confusing to explain it that way (`--undo/backout-in-descendants`). Maybe `--restore-descendants` would work? The idea is that it matches the `jj restore` semantics.

Reply:
Another point I had read was the `--restore-descendants` could not be later abbreviated as `-r` or `-R` since this is used already, but nothing prevents us from using another letter, that is not a rare practice.

I'll update the PR with `--restore-descendants` and see how it feels.